treeherder/perf/sheriffing_criteria/criteria_tracking.py

```diff
@@ -0,0 +1,328 @@
+import csv
+from dataclasses import dataclass, asdict, replace, fields
+import logging
+from multiprocessing import cpu_count
+from multiprocessing.pool import Pool, ThreadPool, AsyncResult
+import time
+from typing import Tuple, Dict, Union, List
+
+from datetime import datetime, timedelta
+
+from treeherder.perf.exceptions import NoFiledBugs
+from .bugzilla_formulas import BugzillaFormula, EngineerTractionFormula, FixRatioFormula
+from treeherder.utils import PROJECT_ROOT
+
+CRITERIA_FILENAME = 'perf-sheriffing-criteria.csv'
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class CriteriaRecord:
+    Framework: str
+    Suite: str
+    EngineerTraction: Union[float, str]
+    FixRatio: Union[float, str]
+    LastUpdatedOn: datetime
+    ShouldSync: bool
```

Comment:
As discussed, `ShouldSync` may suggest that it will force sync if the value is true, so it seems better to have it `AllowSync`  (as suggested by Andra)